WormsApp/board.h

```diff
@@ -13,15 +14,23 @@ enum WormType {
 class Board {
  protected:
   std::vector<std::vector<int>> board_;
-  std::unordered_map<int, std::thread> worms_;
+  //std::unordered_map<int, std::thread> worms_;
+  MapperWrapper<int, std::thread> worms_;
   std::unordered_map<int, WormType> wormTypes_;
   std::unordered_set<int> killed_;
   int nextId_;
   bool killAll_;
+  std::mutex mut;
+
 
- public:
+public:
   Board(int width, int height);
-  virtual ~Board() { }
+
+  std::unordered_map<int, std::thread> getWorms()
+  {
+      return std::move (worms_.mapper);
```

Comment:
I don't understand this - is it needed? You destroy the mapper inside `worms_` while whole board class can still live. I think it may cause problems.

Reply:
Removed, getMapper() method is using instead.